pkg/controller/provider/web/base/client.go

```diff
@@ -23,15 +24,59 @@ var Settings = &settings.Settings
 // Errors
 var (
 	ResourceNotResolvedErr = errors.New("resource (kind) not resolved")
+	RefNotUniqueErr        = errors.New("ref matched multiple resources")
+	NotFoundErr            = errors.New("not found")
 )
 
+//
+// Reference.
+type Ref = ref.Ref
+
 //
 // Resolves resources to API paths.
 type Resolver interface {
 	// Find the API path for the specified resource.
 	Path(resource interface{}, id string) (string, error)
 }
 
+//
+// Resource Finder.
+type Finder interface {
+	// Find a resource by ref.
+	// Returns:
+	//   ProviderNotSupportedErr
+	//   ProviderNotReadyErr
+	//   NotFoundErr
+	//   RefNotUniqueErr
+	ByRef(client Client, resource interface{}, ref Ref) error
+}
+
+//
+// REST Client.
+type Client interface {
+	// Get a resource.
+	// The `resource` must be a pointer to a resource object.
+	// Returns:
+	//   ProviderNotSupportedErr
+	//   ProviderNotReadyErr
+	//   NotFoundErr
+	Get(resource interface{}, id string) error
+	// List a collection.
+	// The `list` must be a pointer to a slice of resource object.
+	// Returns:
+	//   ProviderNotSupportedErr
+	//   ProviderNotReadyErr
+	//   NotFoundErr
+	List(list interface{}, param ...Param) error
+	// Get a resource by ref.
+	// Returns:
+	//   ProviderNotSupportedErr
+	//   ProviderNotReadyErr
+	//   NotFoundErr
+	//   RefNotUniqueErr
+	Find(resource interface{}, ref Ref) error
```

Comment:
For convenience, added the `Find()` method to support finding resources using the `Ref` (id, name, type).